File: problems/121. Best Time to Buy and Sell Stock/Solution.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int buyIndex = 0, maxProfit = 0;
        for (size_t i = 0; i < prices.size(); i++) {
            if (prices[i] < prices[buyIndex]) buyIndex = i;
            maxProfit = max(prices[i] - prices[buyIndex], maxProfit);
        }

        return maxProfit > 0 ? maxProfit : 0;
    }
};

File: problems/121. Best Time to Buy and Sell Stock/Solution.java
class Solution {
    public int maxProfit(int[] prices) {
        int buyIndex = 0, profit = 0;
        for (int i = 0; i < prices.length; i++) {
            if (prices[i] < prices[buyIndex]) buyIndex = i;
            profit = Math.max(profit, prices[i] - prices[buyIndex]);
        }
        return profit > 0 ? profit : 0;
    }
}